#!/usr/bin/env python
PACKAGE = "xaxxon_openlidar"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("header_offset", double_t, 0, "scan header offset from photo sensor, degrees", 189.0,  0.0, 360.0)
gen.add("masks", str_t, 0, "frame masks CW/RHR+ start-mask/stop-mask pairs, degrees 0-360, space separated", "54 80  162 174  190 208  279 305") # TODO: set as standalone default
gen.add("minimum_range", double_t, 0, "minimum output for laser sensor, meters", 0.5,  0.0, 40.0) 
gen.add("maximum_range", double_t, 0, "maximum output for laser sensor, meters", 40.0,  0.0, 40.0) 
gen.add("rpm", int_t, 0, "rev speed for lidar motor, revs per minute", 220, 60,  250) 
gen.add("dropscan_turnrate", int_t, 0, "blank scans when exceeding mobile base odom yaw rate, deg per sec, 0 to disable", 45, 0,  360) 

exit(gen.generate(PACKAGE, "xaxxon_openlidar", "XaxxonOpenLidar"))
